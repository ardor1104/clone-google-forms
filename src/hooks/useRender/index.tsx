import { BrowserRouter } from 'react-router-dom';

import renderWithProviders from 'utils/renderWithProviders';

import { ChildrenPropType } from 'components/globalType';
import ThemeTemplate from 'components/templates/ThemeTemplate';

const useRender = (children: ChildrenPropType) => {
  return renderWithProviders(
    <BrowserRouter>
      <ThemeTemplate>{children}</ThemeTemplate>
    </BrowserRouter>,
  );
};

export default useRender;
