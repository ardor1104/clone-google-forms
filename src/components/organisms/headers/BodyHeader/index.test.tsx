import { composeStories } from '@storybook/testing-react';

import useRender from 'hooks/useRender';

import * as stories from './index.stories';

const { Default } = composeStories(stories);

describe('organisms/headers/BodyHeader', () => {
  it('render Default', async () => {
    const component = useRender(<Default />);

    expect(component).toMatchSnapshot();
  });
});
