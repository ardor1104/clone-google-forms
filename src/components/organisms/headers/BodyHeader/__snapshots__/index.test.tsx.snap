// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/headers/BodyHeader render Default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-ftTHYK sc-pyfCe kCaRnw iLqwPp"
      >
        <div
          class="sc-jrcTuL bKScqn"
        >
          <div
            class="sc-kDvujY sc-ipEyDJ kcNEZW eoqSlP"
          >
            <button
              class="sc-dkrFOg jOmIxm sc-eDvSVe iscdyn"
              kind="none"
              role="button"
            >
              <span
                class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                width="24"
              >
                <svg
                  fill="none"
                  focusable="false"
                />
              </span>
            </button>
            <div
              class="sc-csuSiG fPnGPY"
            >
              <span
                class="sc-hLBbgP cfuRrK"
                width="40"
              >
                <svg
                  fill="none"
                  focusable="false"
                />
              </span>
              <div
                class="sc-eDWCr fMkCcN"
              >
                설문지
              </div>
            </div>
          </div>
          <div
            class="sc-bqWxrE iRLCHi"
          >
            <div
              class="sc-gKPRtg exSMDh"
            >
              <button
                class="sc-dkrFOg jOmIxm sc-eDvSVe kEYClL"
                kind="none"
                role="button"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
              </button>
              <input
                class="sc-iBYQkv iCdRjF"
                placeholder="검색"
                value=""
              />
              <button
                class="sc-dkrFOg jOmIxm sc-eDvSVe edDhAU"
                kind="none"
                role="button"
              >
                <span
                  class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                  width="24"
                >
                  <svg
                    fill="none"
                    focusable="false"
                  />
                </span>
              </button>
            </div>
          </div>
          <div
            class="sc-kDvujY sc-ksBlkl kcNEZW eXhMpo"
          >
            <button
              class="sc-dkrFOg jOmIxm sc-eDvSVe gnpfWU"
              kind="none"
              role="button"
            >
              <span
                class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                width="24"
              >
                <svg
                  fill="none"
                  focusable="false"
                />
              </span>
            </button>
            <div
              class="sc-hBxehG kfuNDB"
            >
              <img
                alt="ProfileImage"
                class="sc-fnGiBr doHTgW"
                src="https://www.truebluenation.com/forum/uploads/monthly_2017_09/T.png.b6d03c176342eab9a77402726a051291.png"
              />
            </div>
          </div>
        </div>
      </header>
      <div
        class="sc-ftTHYK kCaRnw"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-ftTHYK sc-pyfCe kCaRnw iLqwPp"
    >
      <div
        class="sc-jrcTuL bKScqn"
      >
        <div
          class="sc-kDvujY sc-ipEyDJ kcNEZW eoqSlP"
        >
          <button
            class="sc-dkrFOg jOmIxm sc-eDvSVe iscdyn"
            kind="none"
            role="button"
          >
            <span
              class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
              width="24"
            >
              <svg
                fill="none"
                focusable="false"
              />
            </span>
          </button>
          <div
            class="sc-csuSiG fPnGPY"
          >
            <span
              class="sc-hLBbgP cfuRrK"
              width="40"
            >
              <svg
                fill="none"
                focusable="false"
              />
            </span>
            <div
              class="sc-eDWCr fMkCcN"
            >
              설문지
            </div>
          </div>
        </div>
        <div
          class="sc-bqWxrE iRLCHi"
        >
          <div
            class="sc-gKPRtg exSMDh"
          >
            <button
              class="sc-dkrFOg jOmIxm sc-eDvSVe kEYClL"
              kind="none"
              role="button"
            >
              <span
                class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                width="24"
              >
                <svg
                  fill="none"
                  focusable="false"
                />
              </span>
            </button>
            <input
              class="sc-iBYQkv iCdRjF"
              placeholder="검색"
              value=""
            />
            <button
              class="sc-dkrFOg jOmIxm sc-eDvSVe edDhAU"
              kind="none"
              role="button"
            >
              <span
                class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
                width="24"
              >
                <svg
                  fill="none"
                  focusable="false"
                />
              </span>
            </button>
          </div>
        </div>
        <div
          class="sc-kDvujY sc-ksBlkl kcNEZW eXhMpo"
        >
          <button
            class="sc-dkrFOg jOmIxm sc-eDvSVe gnpfWU"
            kind="none"
            role="button"
          >
            <span
              class="sc-hLBbgP gNfMnh sc-jSUZER fTKRNL"
              width="24"
            >
              <svg
                fill="none"
                focusable="false"
              />
            </span>
          </button>
          <div
            class="sc-hBxehG kfuNDB"
          >
            <img
              alt="ProfileImage"
              class="sc-fnGiBr doHTgW"
              src="https://www.truebluenation.com/forum/uploads/monthly_2017_09/T.png.b6d03c176342eab9a77402726a051291.png"
            />
          </div>
        </div>
      </div>
    </header>
    <div
      class="sc-ftTHYK kCaRnw"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
